### PR TITLE
improve performance of Light volume pass

### PR DESCRIPTION
**Pull request by samcake**
- Enable depth test for light volume pass to avoid working on hidden light volumes
- Remove unnecessary separate calls to clear the deferred buffer, all the targets can be cleaned in one call

Enabling the depth test for light volumes does remove the stutering we were observing in Forum:
https://highfidelity.fogbugz.com/f/cases/499/When-panning-across-spot-lights-in-HMD-we-see-large-spikes-in-rendering-time-and-resulting-judder

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 89f18aee3b54661075204d238e2da5136a4e4c30 available at: https://gist.github.com/b71171e5160239684828e183477fc8c5<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7922.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7922.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 185f06944177888dd54b4e58702e01dd83cafc0a available at: https://gist.github.com/b71171e5160239684828e183477fc8c5<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7922.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7922.dmg

**Comment by samcake** [comment]
Test Plans:
Everything should look the same, specifically regarding point and spot lights
the bug was reproduced in Forum, in the dressing room, with a HMD, looking around there was a stuterring occuring (looking at the sign on the left of the room) this stuterring should now be gone.

File: libraries/render-utils/src/DeferredLightingEffect.cpp [hunk 1]
```diff
@@ -548,14 +549,13 @@ static void loadLightProgram(const char* vertSource, const char* fragSource, boo
...
     if (lightVolume) {
         state->setCullMode(gpu::State::CULL_BACK);
-        
-        // No need for z test since the depth buffer is not bound state->setDepthTest(true, false, gpu::LESS_EQUAL);
-        // TODO: We should bind the true depth buffer both as RT and texture for the depth test
+        state->setDepthTest(true, false, gpu::LESS_EQUAL);
+
         // TODO: We should use DepthClamp and avoid changing geometry for inside /outside cases
-        state->setDepthClampEnable(true);
+        //state->setDepthClampEnable(true);
```
Comment (ZappoMan) [inline libraries/render-utils/src/DeferredLightingEffect.cpp]: is this dead code? Can we remove it?

File: libraries/render-utils/src/DeferredLightingEffect.cpp [hunk 2]
```diff
@@ -662,10 +662,14 @@ model::MeshPointer DeferredLightingEffect::getSpotLightMesh() {
         _spotLightMesh->setIndexBuffer(gpu::BufferView(new gpu::Buffer(sizeof(unsigned short) * indices, (gpu::Byte*) indexData), gpu::Element::INDEX_UINT16));
         delete[] indexData;
 
-        model::Mesh::Part part(0, indices, 0, model::Mesh::TRIANGLES);
-        //DEBUG: model::Mesh::Part part(0, indices, 0, model::Mesh::LINE_STRIP);
         
-        _spotLightMesh->setPartBuffer(gpu::BufferView(new gpu::Buffer(sizeof(part), (gpu::Byte*) &part), gpu::Element::PART_DRAWCALL));
+        std::vector<model::Mesh::Part> parts;
+        parts.push_back(model::Mesh::Part(0, indices, 0, model::Mesh::TRIANGLES));
+        //DEBUG:
```
Comment (ZappoMan) [inline libraries/render-utils/src/DeferredLightingEffect.cpp]: is this dead code? Can we remove it?

File: libraries/render-utils/src/DeferredLightingEffect.cpp [hunk 3]
```diff
@@ -469,9 +464,15 @@ void DeferredLightingEffect::render(const render::RenderContextPointer& renderCo
...
 
-                        auto& part = mesh->getPartBuffer().get<model::Mesh::Part>();
-
-                        batch.drawIndexed(model::Mesh::topologyToPrimitive(part._topology), part._numIndices, part._startIndex);
+                        {
+                            auto& part = mesh->getPartBuffer().get<model::Mesh::Part>(0);
+                            batch.drawIndexed(model::Mesh::topologyToPrimitive(part._topology), part._numIndices, part._startIndex);
+                        }
+                        // keep for debug
+                   /*     {
```
Comment (ZappoMan) [inline libraries/render-utils/src/DeferredLightingEffect.cpp]: can we clean this up a little bit... maybe an `#ifdef`

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2d7118dacc87260faca4f232727823ca7b8985d2 available at: https://gist.github.com/b71171e5160239684828e183477fc8c5<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7922.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7922.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 039bb3ac3cda2dee8135d2acc42e840425733f21 available at: https://gist.github.com/b71171e5160239684828e183477fc8c5<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7922.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7922.dmg

**Comment by OmegaHeron** [comment]
QA Pass for desktop mode only - Needs HMD test.
Testing Windows 10 64 bit - i5 6500 CPU - GTX 970 GPU - **Desktop mode - keyboard/mouse**
All standard tests pass.  I can see no difference in standard home scene between this PR and RC Interface (as indicated should be case).

**Comment by samcake** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 039bb3ac3cda2dee8135d2acc42e840425733f21 available at: https://gist.github.com/b71171e5160239684828e183477fc8c5<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7922.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7922.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 67929a018c90571c4ee4b017e68fb0da47dd3101 available at: https://gist.github.com/b71171e5160239684828e183477fc8c5<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7922.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7922.dmg

**Comment by ZappoMan** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 67929a018c90571c4ee4b017e68fb0da47dd3101 available at: https://gist.github.com/b71171e5160239684828e183477fc8c5<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7922.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7922.dmg

**Comment by ZappoMan** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 67929a018c90571c4ee4b017e68fb0da47dd3101 available at: https://gist.github.com/ca94503f3826980c9ece8b493a5895ed<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7922.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7922.dmg

**Comment by ZappoMan** [comment]
it looks like the failures were caused by failures to upload symbols to bugsplat

**Pull request merged**